src/AppInstallerCLICore/Workflows/PortableInstallFlow.cpp

```diff
@@ -0,0 +1,309 @@
+        Manifest::ScopeEnum scope = ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope));
+        Utility::Architecture arch = context.Get<Execution::Data::Installer>()->Arch;
+
+        std::filesystem::path targetInstallDirectory;
+
+        if (!locationArg.empty())
+        {
+            targetInstallDirectory = std::filesystem::path{ ConvertToUTF16(locationArg) };
+        }
+        else
+        {
+            targetInstallDirectory = GetPortableInstallRoot(scope, arch);
+        }
+
+        targetInstallDirectory /= packageId;
+
+        std::filesystem::create_directories(targetInstallDirectory);
+
+        std::string fileName;
+        if (!renameArg.empty())
+        {
+            fileName = Normalize(renameArg);
+        }
+        else
+        {
+            fileName = installerPath.filename().u8string();
+        }
+
+        AppendExeExtension(fileName);
+
+        std::filesystem::path targetInstallFullPath = targetInstallDirectory / fileName;
+        return targetInstallFullPath;
+    }
+
+    void CreatePortableSymlink(Execution::Context& context)
+    {
+        const std::filesystem::path installerPath = context.Get<Execution::Data::InstallerPath>();
+        Manifest::ScopeEnum scope = ConvertToScopeEnum(context.Args.GetArg(Execution::Args::Type::InstallScope));
+        Utility::Architecture arch = context.Get<Execution::Data::Installer>()->Arch;
+        std::vector<string_t> commands = context.Get<Execution::Data::Installer>()->Commands;
+        std::string_view renameArg = context.Args.GetArg(Execution::Args::Type::Rename);
+
+        std::string fileName;
+        std::string commandAlias;
+        if (!renameArg.empty())
+        {
+            std::string renameArgValue = Normalize(renameArg);
+            commandAlias = renameArgValue;
+            fileName = renameArgValue;
+        }
+        else
+        {
+            commandAlias = !commands.empty() ? commands[0] : installerPath.filename().u8string();
+        }
+
+        AppendExeExtension(commandAlias);
+        std::filesystem::path symlinkPath = GetPortableLinksLocation(scope, arch) / commandAlias;
+        std::filesystem::path portableTargetFullPath = GetPortableTargetFullPath(context);
+        AppInstaller::Filesystem::CreateSymlink(portableTargetFullPath, symlinkPath);
+    }
+
+    std::optional<HRESULT> PortableCopyExeInstall(Execution::Context& context, IProgressCallback& progress)
+    {
+        const std::filesystem::path installerPath = context.Get<Execution::Data::InstallerPath>();
+        std::string_view renameArg = context.Args.GetArg(Execution::Args::Type::Rename);
+
+        std::string fileName;
+        if (!renameArg.empty())
+        {
+            std::string renameArgValue = Normalize(renameArg);
+            fileName = renameArgValue;
+        }
+        else
+        {
+            fileName = installerPath.filename().u8string();
+        }
+
+        AppendExeExtension(fileName);
```

Comment:
I think you forgot to remove this `fileName` calculation block in favor of `GetPortableTargetFullPath`

Reply:
Removed filename calculation block